import_specifications/templates/README.md

```diff
@@ -0,0 +1,74 @@
+import cell in the Narrative. The user would fill in rows 4 and above (2 and above in Excel)
+in the templates with the same data they would use to fill in the UI. For checkboxes, specify
+`1` for checked and `0` for unchecked.
+
+There is one caveat, in that the templates currently support specifying all the parameters for
+an import for each import, while the UI does not. Currently the first row of data in the template
+will be used to specify parameters for all the other rows for the case where the UI
+does not allow individual parameter settings per row (see note 1). Only input files and output
+object names are allowed to differ for a set of imports for a type.
+
+Once the template or templates are filled out, upload the templates and the corresponding files
+to be imported to KBase to the narrative staging area. For each template, choose the
+`Import Specification` type and ensure the selection checkbox for each file is checked. There is
+no need to choose types for or select the files that are specified in the templates - indeed
+selecting those files will cause them to be, in effect, selected twice.
+
+When the template's types have been properly chosen and the selection checkboxes are correct,
+click the `Import Selected` button. The bulk import cell will be created in your narrative and
+populated as per the contents of your selected templates.
+
+## Lookahead inputs
+
+Some importers, such as the Genbank importer, have input fields that are based on a dynamic
+lookup from user input - for example, the taxon ID of a genome is looked up from the prefix
+of a scientific name provided by the user. Clearly specifying a prefix in the template will not
+work in this case - the user must specify the entirety of the input; for instance the entire
+scientific name. To to this, start an import and use the lookup box to find the scientific name
+of interest, and then add that name to the template.
+
+We are considering ways to make this easier in the future.
+
+## Troubleshooting
+
+* Do not change the first 3 rows of the TSV or CSV file or the first row or hidden rows of
+  Excel files. This will cause errors or incorrect data.
+* Be careful of stray cells in Excel - ensure that all the cells outside your data block are
+  empty. Consider deleting all the rows and columns outside your data block.
+* For CSV and TSV files, be careful to always include the correct number of separator characters
+  for every row, even if some values are blank. The importer is picky about this to help the user
+  avoid off-by-one (or more) errors in their data.
+* Only one file or Excel tab is allowed per data type. Do not include rows for the same data type
+  in more than one Excel tab, more than one file, a tab and an Excel file, etc.
+* In CSV and TSV files, white space around non-numerical data is ignored, and whitespace prior to
+  numerical data is ignored.
+* In CSV and TSV files, you can use quotes (`"`) to surround text that contains the separator
```

Comment:
Should this be allowed or encouraged? I assumed both would be illegal characters anyway.

Reply:
This is just generally how most CSV writing and parsing programs work, but you're right, this isn't a case users should come across for the current apps (except maybe scientific names?) so maybe we could delete it. What do you think?